utils/memory_bench/src/main.rs

```diff
@@ -0,0 +1,421 @@
+        path.push(&component);
+        if !path.exists() {
+            panic!("Could not find the component at the path: {:?}", path);
+        }
+        path
+    };
+
+    let example_dir = {
+        let mut path = component_dir.clone();
+        path.push("examples");
+        if !path.exists() {
+            panic!(
+                "Could not find the examples directory at the path: {:?}",
+                path
+            );
+        }
+        path
+    };
+
+    let benchmark_dir = {
+        let mut path = root_dir;
+        path.push("benchmarks/memory");
+        path.push(os);
+        path.push(&component);
+        path
+    };
+
+    let benchmark_output_path = {
+        let mut path = benchmark_dir.clone();
+        path.push("output.ndjson");
+        path
+    };
+
+    fs::create_dir_all(&benchmark_dir).unwrap_or_else(|err| {
+        panic!(
+            "Unable to create the benchmark directory {:?} {:?}",
+            benchmark_dir, err
+        );
+    });
+
+    if benchmark_output_path.exists() {
+        fs::remove_file(&benchmark_output_path).unwrap_or_else(|err| {
+            panic!(
+                "Could not remove the file: {:?} {:?}",
+                benchmark_output_path, err
+            );
+        });
+    }
+
+    let mut benchmark_output = fs::OpenOptions::new()
+        .read(true)
+        .append(true)
+        .create(true)
+        .open(&benchmark_output_path)
+        .expect("Unable to open the benchmark output file for write.");
+
+    for example in examples {
+        println!("[memory_bench] Starting example {:?}", example);
+        let example_path = {
+            let mut path = example_dir.clone();
+            path.push(format!("{}.rs", example));
+            if !path.exists() {
+                eprintln!("Could not find the example at the path: {:?}", path);
+                process::exit(1);
+            }
+            path
+        };
+        let example_copy_path = {
+            let mut path = example_dir.clone();
+            path.push(format!("{}___memory.rs", example));
+            path
+        };
+
+        println!(
+            "[memory_bench] Injecting the memory instrumentation into {:?}",
+            example
+        );
+        let instrumented_src = run_dhat_injection(&example_path);
+
+        let mut file = File::create(&example_copy_path).unwrap_or_else(|err| {
+            panic!(
+                "Could not create the file {:?} {:?}",
+                example_copy_path, err
+            );
+        });
+
+        file.write_all(instrumented_src.as_bytes())
+            .unwrap_or_else(|err| {
+                eprintln!("Could not write the file {:?} {:?}", example_copy_path, err);
+                process::exit(1);
+            });
+
+        println!(
+            "[memory_bench] Running the instrumented example {:?}",
+            example
+        );
+
+        let mut run_example = Command::new("cargo")
+            .arg("+nightly")
```

Comment:
Why do you need `+nightly`?

Reply:
The `--profile bench` is an unstable command line flag, which requires the nightly toolchain. I believe I documented this in the CI script, but should probably add a note here. The custom profile is needed for "release" optimizations, but with debug information.